scripts/finalizeCCTPMessages.ts

```diff
@@ -0,0 +1,309 @@
+import { getContractFactory, ethers, SignerWithAddress } from "../utils/utils";
+import { hre } from "../utils/utils.hre";
+import { L1_ADDRESS_MAP, L2_ADDRESS_MAP, CIRCLE_DOMAIN_IDs } from "../deploy/consts";
+import { getNodeUrl } from "@uma/common";
+import { Event, Wallet, providers } from "ethers";
+import fetch from "node-fetch";
+
+const MAX_L1_BLOCK_LOOKBACK = 1000;
+const MAX_L2_BLOCK_LOOKBACK = 1000;
+
+const chainToArtifactPrefix: Record<number, string> = {
+  1: "Ethereum",
+  5: "Ethereum",
+  10: "Optimism",
+  420: "Optimism",
+  42161: "Arbitrum",
+  421613: "Arbitrum",
+  8453: "Base",
+  84531: "Base",
+  137: "Polygon",
+  80001: "Polygon",
+};
+
+const messageTransmitterAbi = [
+  {
+    inputs: [
+      {
+        internalType: "bytes",
+        name: "message",
+        type: "bytes",
+      },
+      {
+        internalType: "bytes",
+        name: "attestation",
+        type: "bytes",
+      },
+    ],
+    name: "receiveMessage",
+    outputs: [
+      {
+        internalType: "bool",
+        name: "success",
+        type: "bool",
+      },
+    ],
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+  {
+    inputs: [
+      {
+        internalType: "bytes32",
+        name: "",
+        type: "bytes32",
+      },
+    ],
+    name: "usedNonces",
+    outputs: [
+      {
+        internalType: "uint256",
+        name: "",
+        type: "uint256",
+      },
+    ],
+    stateMutability: "view",
+    type: "function",
+  },
+];
+
+/**
+ * Script to claim L1->L2 or L2->L1 messages. Run via
+ * ```
+ * RECEIVE_MESSAGES_ON=l1 \
+ * yarn hardhat run ./scripts/claimLineaMessages.ts \
+ * --network linea-goerli \
+ * ```
+ * Environment variables:
+ * - `RECEIVE_MESSAGES_ON`: Which messages to claim. Either `l1` or `l2`.
+ * Flags:
+ * - `--network`: The L2 network, i.e. `linea-goerli` or `linea`.
```

Comment:
```suggestion
 * - `--network`: The L2 network, i.e. `arbitrum-goerli` or `arbitrum`.
```
Linea doesn't have CCTP deployed (yet)